warehouse/router_scheduling.go

```diff
@@ -7,108 +7,69 @@ import (
 	"sync"
 	"time"
 
-	"github.com/cenkalti/backoff/v4"
 	"github.com/samber/lo"
 
-	"github.com/rudderlabs/rudder-go-kit/config"
 	"github.com/rudderlabs/rudder-server/utils/timeutil"
 	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
 	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 )
 
 var (
 	scheduledTimesCache     map[string][]int
-	minUploadBackoff        time.Duration
-	maxUploadBackoff        time.Duration
-	startUploadAlways       bool
 	scheduledTimesCacheLock sync.RWMutex
+
+	startUploadAlways bool
```

Comment:
There might be a race here :thinking: When is the `Admin.TriggerUpload()` fired? Can it run concurrently with `Router.canCreateUpload()`?

Reply:
Right, it shouldn't be much of a problem then.

Reply:
I'd prefer correctness over a potential benign data race ([if such things exist](https://bartoszmilewski.com/2020/08/11/benign-data-races-considered-harmful/)). Converting this to an `atomic.Bool` should be a no :brain:

Reply:
Good point.